### ISSUE TITLE
Never modify actual database while trimming

### ISSUE THREAD

**Issue by gmemstr** [gmemstr/database-janitor#16]
Rework architecture to allow the ability to NOT modify database _ever_ when trimming. 

It looks like the MySQLdump-php library already accepts the ability to not dump out the contents of a table without queries so probably the route for scrubs, might need to be slightly modified to match our trim routine (will work fine for scrubbing).

**Comment by gmemstr** [comment]
While this issue is semi-resolved, there is some more work that can be done to the MySQLdump library to further reduce calls SQL queries on our end. Specifically, selecting rows into "no data" tables to avoid creating new ones and renaming them on the fly.

**Comment by gmemstr** [comment]
We have a couple options with this, all with their own trade offs and advantages.

 1. Use the built-in "where" flag in MySQLdump-php
    - Unfortunately this reads as if it will only select where this column is correct for ALL tables - this isn't ideal since for big databases we will likely miss data in tables that don't have the col/value when we actually want to keep it. 

```php
$stmt = "SELECT ".implode(",", $colStmt)." FROM `$tableName`";

if ($this->dumpSettings['where']) {
  $stmt .= " WHERE {$this->dumpSettings['where']}";
}
```

 2. Add a new "keep-data" flag to MySQLdump-php library to allow more fine grained control
    - This would likely add additional configuration file complexity (which could be somewhat addressed with #18) and I'm still not 100% sure how to tackle this.

**Comment by gmemstr** [comment]
See cd90726f8ffe30c523fd2e8779768d7b7af52cfc
